### PR TITLE
Adding the HashTable in Data Structures

### PR DESCRIPTION
**Pull request by Ravikisha**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: DataStructures/Hashing/NumberTheory/PrimeNumber.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        /// <summary>
+        /// Gets the next prime number.
+        /// </summary>
+        /// <param name="number">Number to start from.</param>
+        /// <param name="factor">Factor to multiply the number by.</param>
+        /// <param name="desc">True to get the previous prime number, false otherwise.</param>
+        /// <returns>The next prime number.</returns>
+        public static int NextPrime(int number, int factor = 1, bool desc = false)
```
Comment (siriak) [inline DataStructures/Hashing/NumberTheory/PrimeNumber.cs:55]: What is `factor` here? Why do you need it?

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix PR checks

File: DataStructures/Hashing/NumberTheory/PrimeNumber.cs (hunk 1 above)
Comment (Ravikisha) [reply DataStructures/Hashing/NumberTheory/PrimeNumber.cs:55]: the factor parameter allows you to start the search for the next prime number from a value that is a multiple of number. It influences the step size by which the function searches for the next prime, which can be useful in certain situations where you need to find prime numbers that have specific relationships with other numbers.

File: DataStructures/Hashing/HashTable.cs [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            {
+                throw new ArgumentOutOfRangeException(nameof(loadFactor), "Load factor must be greater than 0");
+            }
+
+            if (loadFactor > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(loadFactor), "Load factor must be less than or equal to 1");
+            }
+
+            this.capacity = PrimeNumber.NextPrime(capacity);
```
[the remark this answers is not in the document]
Comment (Ravikisha) [reply DataStructures/Hashing/HashTable.cs:122]: `PrimeSquence` is part of the Algorithm Project and If I add the Algorithm as a dependency then it creates a circular dependency which is not allowed.

File: DataStructures/Hashing/NumberTheory/PrimeNumber.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    /// Class for prime number operations.
+    /// </summary>
+    /// <remarks>
+    /// A prime number is a natural number greater than 1 that is not a product of two smaller natural numbers.
+    /// </remarks>
+    public class PrimeNumber
+    {
+        public PrimeNumber()
+        {
+            Console.WriteLine("PrimeNumber");
```
Comment (siriak) [inline DataStructures/Hashing/NumberTheory/PrimeNumber.cs:17]: Please remove logging to console

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix these Codacy issues, everything else looks good
image

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**